src/Microsoft.Health.SqlServer/Features/Schema/SchemaInitializer.cs

```diff
@@ -113,14 +114,22 @@ public async Task InitializeAsync(bool forceIncrementalSchemaUpgrade = false, Ca
 
         private async Task GetCurrentSchemaVersionAsync(CancellationToken cancellationToken)
         {
-            int version = await _schemaManagerDataStore.GetCurrentSchemaVersionAsync(cancellationToken);
-            if (version != 0)
+            if (!_canCallGetCurrentSchema)
             {
-                _schemaInformation.Current = version;
+                _canCallGetCurrentSchema = await _schemaManagerDataStore.ObjectExistsAsync("SelectCurrentSchemaVersion", "P", cancellationToken);
             }
-            else
+
+            if (_canCallGetCurrentSchema)
             {
-                _logger.LogInformation("No version found. It must be new database");
+                int version = await _schemaManagerDataStore.GetCurrentSchemaVersionAsync(cancellationToken);
+                if (version != 0)
+                {
+                    _schemaInformation.Current = version;
+                }
+                else
+                {
+                    _logger.LogInformation("No version found. It must be new database");
```

Comment:
Would we use this same message if `_canCallGetCurrentSchema` is `false`?

Reply:
good point. Added a warning for SP not found